metrics/processor.go

```diff
@@ -122,9 +122,15 @@ func (p *Processor) UptimeMetrics(startTime time.Time) Metrics {
 }
 
 func (p *Processor) CpuMetrics(s *stats.CPUStats, limit float64) Metrics {
+	if s == nil {
+		return nil
+	}
 	// Convert nanoCores to seconds
 	var cpuUsage float64
-	nanoCores := s.UsageNanoCores
+	var nanoCores *uint64
+	if s.UsageNanoCores != nil {
```

Comment:
Is this null-check strictly necessary, if we check for null `nanoCores` below?

Reply:
Unsure. My concern was that setting `nanoCores` equal to `s.UsageNanoCores` would throw an exception if `s.UsageNanoCores` doesn't exist.

Reply:
That should be fine. The assignment will just result in `nanoCores` being `nil`.